orchagent/routeorch.cpp

```diff
@@ -103,6 +103,85 @@ RouteOrch::RouteOrch(DBConnector *db, string tableName, NeighOrch *neighOrch) :
     m_syncdRoutes[v6_default_ip_prefix] = IpAddresses();
 
     SWSS_LOG_NOTICE("Create IPv6 default route with packet action drop");
+
+    /* All the interfaces have the same MAC address and hence the same
+     * auto-generated link-local ipv6 address with eui64 interface-id.
+     * Hence add a single /128 route entry for the link-local interface
+     * address pointing to the CPU port.
+     */
+
+    IpPrefix linklocal_prefix = getLinkLocalEui64Addr();
+
+    addLinkLocalRouteToMe(gVirtualRouterId, linklocal_prefix);
```

Comment:
I think we should also add the fe80::/10

Reply:
Ok. Why do we need the fe80::/10 route? Is it to handle all packets destined to any link-local address(es) belonging to us.

Reply:
@nikos-github, I am just thinking what are the other cases where we might need packets with DIP matching fe80::/10 link-local subnet prefix to come to CPU. I don't think we would need any other link-local packets other than fe80.../128 to come to CPU. Am I missing something here?

Reply:
Agree. Verified that the fe80::/10 subnet route to CPU handles both the scenarios that this PR is raised for (pinging to link-local address and link-local address as nexthop).
Now, we don't need the fe80.../128 route to CPU to be added to hardware since fe80::/10 covers that too. Retained in the code, the function that returns our own interface link-local address with eui64 interface id, for later reference.
@prsunny Since we are no longer adding the fe80.../128 now with this change, i am not currently adding any VS test for the same. Hope that is fine.

Reply:
@kirankella The point I was trying to make is that you may actually want both the /10 and the /128. It really depends on how the hw/SAI is implementing forwarding for the /10 (which btw I do not remember) but in vendor world packets hitting a connected prefix are rate limited when punted to the CPU. However when they hit the receive entry (/32 or /128 added via ip2me) they are not meant to. If SAI's implementation regarding link locals is to not rate limit for the /10, then you're good with the current updated diffs.

Reply:
Agree.
@lguohan @prsunny, In SONiC, I see that, by default the CoPP policy traps applied for ip2me routes rate limits the matching packets to 600 pps. Is this expected?

Reply:
AFAIK, the subnet and ip2me are trapped to the same queue which is currently 600pps. To me, this is expected as we have separate traps for those control packets destined to me, which leaves this trap for packets like ICMP.

Reply:
Yes, the control packets like BGP, LACP, LLDP... and the ARP,NDP packets are going to the same queue (4). And the ARP, NDP packets are rate limited to 600pps. 
Since, subnet and ip2me are trapped to the same queue and are rate limited to 600 pps, currently there does not seem to be any difference in both the routes.
Anyways, we might change the queue mapping between the subnet and ip2me traps in future. So, it is good to have the /128 route too in the hardware for the link-local address (like in case of global interface addresses). @nikos-github @prsunny , do you agree?

Reply:
I'm fine to have both for reasons of functionality in case the underlying SAI implementation changes on how packets are rate limited for the subnet and the host entries, and for consistency purposes.

Reply:
@nikos-github Retained /128 ip2me route too. Thanks.